examples/twittermap/web/public/javascripts/common/services.js

```diff
@@ -312,8 +365,30 @@ angular.module('cloudberry.common', [])
               cloudberryService.timeResult = result.value[0];
               cloudberryService.mapResult = result.value[1];
               cloudberryService.hashTagResult = result.value[2];
-            } else { // this is the {key: "done"} message
-              console.log("receive :", result.value);
+            }
+            // When the query is executed completely
+            else  {
+              MapResultCache.putValues(cloudberryService.parameters.geoIds,
+                cloudberryService.parameters.geoLevel, cloudberryService.mapResult);
+            }
+            break;
+          case "batchWithoutGeoRequest":
+            if(angular.isArray(result.value)) {
+              cloudberryService.timeResult = result.value[0];
+              cloudberryService.hashTagResult = result.value[1];
+            }
+            break;
+          case "batchWithPartialGeoRequest":
+            if(angular.isArray(result.value)) {
+              cloudberryService.timeResult = result.value[0];
+              cloudberryService.mapResult = result.value[1];
+              cloudberryService.hashTagResult = result.value[2];
+            }
+            else  {
+              MapResultCache.putValues(geoIdsNotInCache, cloudberryService.parameters.geoLevel,
+                                      cloudberryService.mapResult);
+              cloudberryService.mapResult = MapResultCache.getValues(cloudberryService.parameters.geoIds,
```

Comment:
we don't need to reset the `mapresult`, right?

Reply:
In line 384, mapresult gets the result for new geoIds and updates the cache store in line 388. In line 390, we get complete results, including ones already in cache.